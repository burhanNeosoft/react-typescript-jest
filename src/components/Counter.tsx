// Counter.tsx
import React, { useState } from 'react';

interface CounterProps {
  initialCount?: number;
}

const Counter: React.FC<CounterProps> = ({ initialCount = 0 }) => {
  const [count, setCount] = useState(initialCount);

  return (
    <div>
      <p>Count: {count}</p>
      <button onClick={() => setCount(count + 1)}>Increment</button>
      <button onClick={() => setCount(count - 1)}>Decrement</button>
    </div>
  );
};

export default Counter;
