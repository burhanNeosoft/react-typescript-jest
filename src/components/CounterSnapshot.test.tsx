import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Counter from './Counter';

describe('Counter', () => {
  it('matches snapshot', () => {
    const { container } = render(<Counter initialCount={5} />);
    expect(container).toMatchSnapshot();
  });
});