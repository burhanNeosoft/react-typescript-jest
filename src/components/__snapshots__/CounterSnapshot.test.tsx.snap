// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches snapshot 1`] = `
<div>
  <div>
    <p>
      Count: 
      5
    </p>
    <button>
      Increment
    </button>
    <button>
      Decrement
    </button>
  </div>
</div>
`;
